ui-src/lib/views/NVRSettingsView.js

```diff
@@ -0,0 +1,174 @@
+// vim: set et sw=2:
+//
+
+import $ from 'jquery';
+
+/**
+ * Class to control the view of NVR Settings.
+ *
+ * These  settings/controls include:
+ * - Max video length
+ * - Trim segment start/end
+ * - Time Format
+ */
+export default class NVRSettingsView {
+  /**
+   * Construct based on element ids
+   */
+  constructor({
+    videoLenId = 'split',
+    trimCheckId = 'trim',
+    tsTrackId = 'ts',
+    timeFmtId = 'timefmt',
+  } = {}) {
+    this._ids = {videoLenId, trimCheckId, tsTrackId, timeFmtId};
+    this._videoLength = null;
+    this._videoLengthHandler = null;
+    this._trim = null;
+    this._trimHandler = null;
+    this._timeFmtStr = null;
+    this._timeFmtHandler = null;
+    this._tsTrack = null;
+    this._tsTrackHandler = null;
+    this._wireControls();
+  }
+
+  /**
+   * Find selected option in <select> and return value, or first option's value.
+   *
+   * The first option's value is returned if no option is selected.
+   *
+   * @param  {jQuery} selectEl jQuery element for the <select>
+   * @return {String}          Value of the selected/first option
+   */
+  _findSelectedOrFirst(selectEl) {
+    let value = selectEl.find(':selected').val();
+    if (!value) {
+      value = selectEl.find('option:first-child').val();
+    }
+    return value;
+  }
+
+  /**
+   * Wire up all controls and handlers.
+   *
+   */
+  _wireControls() {
+    const videoLengthEl = $(`#${this._ids.videoLenId}`);
+    this._videoLength = this._findSelectedOrFirst(videoLengthEl);
+    videoLengthEl.change((e) => {
+      this._videoLength = Number(e.target.value);
```

Comment:
Previously, the empty string was special-cased for infinity. Looks like it turns into 0 now, so it's the opposite extreme from what was intended.

Reply:
That was well hidden... I have used the text "infinite" that more clearly expresses the intent, and now handle that case in the conversion to Number to product Infinity. In the URL forming that is checked for and if specified, the split parameter is left out. This seem to produce the originally intended effect.